test/e2e/pv_reclaimpolicy.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/clientset_generated/clientset"
+	vsphere "k8s.io/kubernetes/pkg/cloudprovider/providers/vsphere"
+	"k8s.io/kubernetes/test/e2e/framework"
+)
+
+var _ = framework.KubeDescribe("persistentvolumereclaim", func() {
+	f := framework.NewDefaultFramework("persistentvolumereclaim")
+
+	var (
+		c          clientset.Interface
+		ns         string
+		volumePath string
+		pv         *v1.PersistentVolume
+		pvc        *v1.PersistentVolumeClaim
+	)
+
+	framework.KubeDescribe("persistentvolumereclaim:vsphere", func() {
+
+		BeforeEach(func() {
+			framework.SkipUnlessProviderIs("vsphere")
+			c = f.ClientSet
```

Comment:
L45 - 47 can be moved to the outer `KubeDescribe` since the would be common to all reclaimPolicy tests. The `SkipUnlessProviderIs` above make sense here.